src/main/java/com/intuit/graphql/orchestrator/schema/transform/FederationTransformerPostMerge.java

```diff
@@ -117,20 +120,26 @@ private void validateBaseExtensionCompatibility(EntityMergingContext entityMergi
   }
 
   private void checkFederationFieldDirectives(EntityMergingContext entityMergingContext) {
-      List<FieldDefinition> fieldDefinitions = getFieldDefinitions(entityMergingContext.getTypeExtension());
+      List<FieldDefinition> extFieldDefinitions = getFieldDefinitions(entityMergingContext.getTypeExtension());
+      List<String> baseFieldNames = getFieldDefinitions(entityMergingContext.getBaseType())
+              .stream().map(FieldDefinition::getName).collect(Collectors.toList());
 
+      extFieldDefinitions.forEach( fieldDefinition -> {
+          boolean sharedField = baseFieldNames.contains(fieldDefinition.getName());
+          AtomicBoolean hasExternalAtomicBoolean = new AtomicBoolean(false);
```

Comment:
curios why use `AtomicBoolean` instead of boolean?

Reply:
variable needed to be effectively final since it is inside a lambda expression. It wouldn't compile if it was boolean or Boolean.

Reply:
got it..